src/window/gui/ConsoleWindow.cpp

```diff
@@ -533,23 +591,29 @@ int ConsoleWindow::CallbackStub(ImGuiInputTextCallbackData* data) {
             if (emptyHistory) {
                 break;
             }
-            if (historyIndex < static_cast<int>(instance->mHistory.size()) - 1) {
-                instance->mHistoryIndex += 1;
+            if (historyIndex > 0) {
```

Comment:
reading through this i'm finding it's not the easiest thing to keep track of `historyIndex` vs `instance->mHistoryIndex`

i'm thinking maybe renaming `historyIndex` could help but i can't think of a better name off the top of my head

Reply:
the change looks great!

one last question before I reapprove: are there any situations where we were *relying* on the local var not changing?

Reply:
Yeah nothing relies on the local var. Before my changes some of the code used it, but it was undesirable and counter intuitive to the user (read: pressing and arrow key to cycle the autofill, but it not taking affect until pressing the key twice because local var was desynced with instance var)